frontend/packages/kubevirt-plugin/integration-tests/tests/vm.snapshots.scenario.ts

```diff
@@ -0,0 +1,53 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { createResource, deleteResource, click } from '@console/shared/src/test-utils/utils';
+import * as editVMSnapshotsView from '../views/vm.snapshots.view';
+import { saveButton, deleteButton } from '../views/kubevirtUIResource.view';
+import { VM_CREATE_AND_EDIT_TIMEOUT_SECS } from './utils/constants/common';
+import { VirtualMachine } from './models/virtualMachine';
+import { getVMManifest } from './mocks/mocks';
+import { getRandStr } from './utils/utils';
+import { READY } from '../../src/utils/strings';
+
+describe('KubeVirt VM Snapshots', () => {
+  const testVM = getVMManifest('Container', testName, `snapshotresourcevm-${getRandStr(5)}`);
+  const vm = new VirtualMachine(testVM.metadata);
+
+  beforeAll(async () => {
+    createResource(testVM);
+  });
+
+  afterAll(() => {
+    deleteResource(testVM);
```

Comment:
shouldn't we ensure that the snapshot gets deleted, for example when the status doe not get ready in time? We could use the leakableResources construct like other tests do

Reply:
Hi, there is also a cypress helper method:  [cy.resourceShouldBeDeleted(testName, resource, name);](https://github.com/openshift/console/blob/66a08e0c120137138888ca879cd7cd7bd70b1b75/frontend/packages/integration-tests-cypress/support/resources.ts#L32).   `cy.resourceShouldBeDeleted` test thru ci that resource is either not found or has been 'marked for deletion' since e2e tests shouldn't have to actually wait for removal of resource.